### PR TITLE
Hediet/september

### PR DESCRIPTION
**Pull request by hediet**

### CODE REVIEW

File: extensions/git/src/commands.ts [hunk 1]
```diff
@@ -1134,6 +1134,48 @@ export class CommandCenter {
...
+	}
+
+	@command('git.runGitMergeDiff3')
+	async runGitMergeDiff3(): Promise<void> {
+		await this.runGitMerge(true);
+	}
+
+	private async runGitMerge(diff3: boolean): Promise<void> {
+		const { activeTab } = window.tabGroups.activeTabGroup;
+		if (!activeTab) {
```
Comment (jrieken) [inline extensions/git/src/commands.ts:1149]: use `instance of vscode.TabInputTextMerge`

File: src/vs/workbench/contrib/mergeEditor/browser/view/editors/resultCodeEditorView.ts [hunk 2]
```diff
@@ -160,5 +168,19 @@ export class ResultCodeEditorView extends CodeEditorView {
...
+		const toolBar = new ToolBar(this.htmlElements.toolbar, contextMenuService);
+		const toolBarUpdate = () => {
+			const secondary: IAction[] = [];
+			createAndFillInActionBarActions(titleMenu, { renderShortTitle: true }, secondary);
+			toolBar.setActions([], secondary);
+		};
+		this._store.add(toolBar);
+		this._store.add(titleMenu);
+		this._store.add(titleMenu.onDidChange(toolBarUpdate));
+		toolBarUpdate();
```
Comment (jrieken) [inline src/vs/workbench/contrib/mergeEditor/browser/view/editors/resultCodeEditorView.ts:184]: This duplicates the menu in the input editors should now move into the super tye

File: src/vs/workbench/services/actions/common/menusExtensionPoint.ts [hunk 3]
```diff
@@ -291,7 +291,11 @@ const apiMenus: IAPIMenu[] = [
 		description: localize('webview.context', "The webview context menu"),
 		proposed: 'contribWebviewContext'
 	},
-
+	{
+		key: 'mergeEditor/result/context',
```
Comment (jrieken) [inline src/vs/workbench/services/actions/common/menusExtensionPoint.ts:295]: Needs proposal, see L292, and needs a correct name. This isn't a context menu but a title-menu, so something like `mergeEditor/result/title`. The fact that it goes into the dropdown/overlfow is the concept of groups

**Review by jrieken — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/mergeEditor/browser/view/editors/resultCodeEditorView.ts (hunk 2 above)
Comment (hediet) [reply src/vs/workbench/contrib/mergeEditor/browser/view/editors/resultCodeEditorView.ts]: I think this is coincidence that all editors use the toolbar, not a semantic dependency. Thus, imo composition makes more sense to de-duplicate the code.

File: src/vs/workbench/services/extensions/common/extensionsApiProposals.ts [hunk 4]
```diff
@@ -37,6 +37,7 @@ export const allApiProposals = Object.freeze({
 	inlineCompletionsNew: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.inlineCompletionsNew.d.ts',
 	interactiveWindow: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.interactiveWindow.d.ts',
 	ipc: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.ipc.d.ts',
+	mergeEditorMenus: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.mergeEditorMenus.d.ts',
```
Comment (jrieken) [inline src/vs/workbench/services/extensions/common/extensionsApiProposals.ts:40]: Contribution only proposals should be prefixed with `contrib`

**Review by jrieken — CHANGES_REQUESTED** [review]

**Review by jrieken — APPROVED** [review]

**Pull request merged**